### ISSUE TITLE
Can not install Flask-Restx in ARM

### ISSUE THREAD

**Issue by FernandoArteaga** [python-restx/flask-restx#356]
## Description:
I am trying to create a docker image in a RaspberryPi which runs `aarch64`, but when the Docker image starts installing the Flask-Restx package, it fails.

!!**This was working correctly 4 months ago because that was the day of my last build.**

## **Environment**
RaspberryPi 4
```commandline
$ uname -a 
Linux RPi 5.10.10-1-ARCH #1 SMP Sat Jan 23 16:26:04 MST 2021 aarch64 GNU/Linux

$ docker --version
Docker version 20.10.2, build 2291f610ae
```

## **requirements.txt**
```text
flask-restx==0.4.0
```

## **Dockerfile**
```dockerfile
# Build
FROM python:3.9 as build

COPY requirements.txt /

WORKDIR /pip-packages

RUN pip download -r /requirements.txt --no-input


# Prod
FROM python:3.9

COPY --from=build /pip-packages/ /pip-packages/

RUN pip install --no-index --find-links=/pip-packages/ /pip-packages/*

RUN pip list
```

## **Expected Behavior**
Docker image can be built correctly.

## **Actual Behavior**
Docker image fails when building.

### **Message Error**
```text
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel
       cwd: None
  Complete output (4 lines):
  Looking in links: /pip-packages/
  Processing /pip-packages/setuptools-57.2.0-py3-none-any.whl
  ERROR: Could not find a version that satisfies the requirement wheel
  ERROR: No matching distribution found for wheel
  ----------------------------------------
WARNING: Discarding file:///pip-packages/pyrsistent-0.18.0.tar.gz. Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel Check the logs for full command output.
ERROR: Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel Check the logs for full command output.
The command '/bin/sh -c pip install --no-index --find-links=/pip-packages/ /pip-packages/*' returned a non-zero code: 1
```

### Docker build output
```commandLine
$ docker build --tag docker-arm-poc-3:latest --no-cache .
```
```text
Sending build context to Docker daemon   5.12kB
Step 1/8 : FROM python:3.9 as build
 ---> 8ff7a2865c18
Step 2/8 : COPY requirements.txt /
 ---> 1f6972b1b83d
Step 3/8 : WORKDIR /pip-packages
 ---> Running in 4d614d4db71b
Removing intermediate container 4d614d4db71b
 ---> 45d2797d6000
Step 4/8 : RUN pip download -r /requirements.txt --no-input
 ---> Running in 53c81d589f32
Collecting flask-restx==0.4.0
  Downloading flask_restx-0.4.0-py2.py3-none-any.whl (5.3 MB)
Collecting six>=1.3.0
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting Flask<2.0.0,>=0.8
  Downloading Flask-1.1.4-py2.py3-none-any.whl (94 kB)
Collecting werkzeug<2.0.0
  Downloading Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting jsonschema
  Downloading jsonschema-3.2.0-py2.py3-none-any.whl (56 kB)
Collecting pytz
  Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
Collecting aniso8601>=0.82
  Downloading aniso8601-9.0.1-py2.py3-none-any.whl (52 kB)
Collecting itsdangerous<2.0,>=0.24
  Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
Collecting Jinja2<3.0,>=2.10.1
  Downloading Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
Collecting click<8.0,>=5.1
  Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting MarkupSafe>=0.23
  Downloading MarkupSafe-2.0.1-cp39-cp39-manylinux2014_aarch64.whl (31 kB)
Collecting setuptools
  Downloading setuptools-57.2.0-py3-none-any.whl (818 kB)
Collecting pyrsistent>=0.14.0
  Downloading pyrsistent-0.18.0.tar.gz (104 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'done'
Collecting attrs>=17.4.0
  Downloading attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Saved ./flask_restx-0.4.0-py2.py3-none-any.whl
Saved ./aniso8601-9.0.1-py2.py3-none-any.whl
Saved ./Flask-1.1.4-py2.py3-none-any.whl
Saved ./click-7.1.2-py2.py3-none-any.whl
Saved ./itsdangerous-1.1.0-py2.py3-none-any.whl
Saved ./Jinja2-2.11.3-py2.py3-none-any.whl
Saved ./MarkupSafe-2.0.1-cp39-cp39-manylinux2014_aarch64.whl
Saved ./six-1.16.0-py2.py3-none-any.whl
Saved ./Werkzeug-1.0.1-py2.py3-none-any.whl
Saved ./jsonschema-3.2.0-py2.py3-none-any.whl
Saved ./attrs-21.2.0-py2.py3-none-any.whl
Saved ./pyrsistent-0.18.0.tar.gz
Saved ./pytz-2021.1-py2.py3-none-any.whl
Saved ./setuptools-57.2.0-py3-none-any.whl
Successfully downloaded flask-restx aniso8601 Flask click itsdangerous Jinja2 MarkupSafe six werkzeug jsonschema attrs pyrsistent pytz setuptools
WARNING: You are using pip version 21.0.1; however, version 21.1.3 is available.
You should consider upgrading via the '/usr/local/bin/python -m pip install --upgrade pip' command.
Removing intermediate container 53c81d589f32
 ---> 1467b30d43ef
Step 5/8 : FROM python:3.9
 ---> 8ff7a2865c18
Step 6/8 : COPY --from=build /pip-packages/ /pip-packages/
 ---> 89c8bfc9a430
Step 7/8 : RUN pip install --no-index --find-links=/pip-packages/ /pip-packages/*
 ---> Running in a10bdd1551cb
Looking in links: /pip-packages/
Processing /pip-packages/Flask-1.1.4-py2.py3-none-any.whl
Processing /pip-packages/Jinja2-2.11.3-py2.py3-none-any.whl
Processing /pip-packages/MarkupSafe-2.0.1-cp39-cp39-manylinux2014_aarch64.whl
Processing /pip-packages/Werkzeug-1.0.1-py2.py3-none-any.whl
Processing /pip-packages/aniso8601-9.0.1-py2.py3-none-any.whl
Processing /pip-packages/attrs-21.2.0-py2.py3-none-any.whl
Processing /pip-packages/click-7.1.2-py2.py3-none-any.whl
Processing /pip-packages/flask_restx-0.4.0-py2.py3-none-any.whl
Processing /pip-packages/itsdangerous-1.1.0-py2.py3-none-any.whl
Processing /pip-packages/jsonschema-3.2.0-py2.py3-none-any.whl
Processing /pip-packages/pyrsistent-0.18.0.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel
       cwd: None
  Complete output (4 lines):
  Looking in links: /pip-packages/
  Processing /pip-packages/setuptools-57.2.0-py3-none-any.whl
  ERROR: Could not find a version that satisfies the requirement wheel
  ERROR: No matching distribution found for wheel
  ----------------------------------------
WARNING: Discarding file:///pip-packages/pyrsistent-0.18.0.tar.gz. Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel Check the logs for full command output.
ERROR: Command errored out with exit status 1: /usr/local/bin/python /usr/local/lib/python3.9/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-qf_d47c3/overlay --no-warn-script-location --no-binary :none: --only-binary :none: --no-index --find-links /pip-packages/ -- 'setuptools>=42' wheel Check the logs for full command output.
The command '/bin/sh -c pip install --no-index --find-links=/pip-packages/ /pip-packages/*' returned a non-zero code: 1
```

**Comment by j5awry** [comment]
This error occurs when attempting to build `pyrsistent`, and is not related to `flask-restx`

Reading deeper, it's because `setuptools` is attempting to build a wheel from `pyrsistent-0.18.0.tar.gz`

To fix, add `wheel` to your requirements.txt to create your local cache. if you keep running into issues building `pyrsistent`, you should hop over to their github for help: https://github.com/tobgu/pyrsistent

**Comment by FernandoArteaga** [comment]
Thank you @j5awry for pointing me in a good direction

**Comment by FernandoArteaga** [comment]
@j5awry was correct in his answer. The issue I am facing is due to the last update of the package `pyrsistent`.
The latest package released on **Jun 28, 2021**, `pyrsistent==0.18.0,` seems to have issues building in ARM architectures.

### HOW TO PREVENT THIS ERROR:
Include in the **requirements.txt** the previus version `0.17.3`:
```text
pyrsistent==0.17.3
flask-restx==0.4.0
```
